### PR TITLE
Add node_tcp target to http_benchmark.

### LINKED ISSUE #1069: wanted: node HTTP benchmark using TCP
Deno does not have an HTTP server currently. It does have a TCP server - and we are interested in optimizing networking / event loop even before the HTTP server has been implemented. 
Thus we fake it:
https://github.com/denoland/deno/blob/master/tests/http_bench.ts

However, the Node program we bench against is actually parsing HTTP. So it's not really a fair battle:
https://github.com/denoland/deno/blob/master/tools/node_http.js

This issue is to write a add a fake HTTP server using Node's `require("net").Server` similar to the `http_bench.ts` above.

### PR DESCRIPTION
**Pull request by ry**
Fixes #1069

cc @kyranet

### CODE REVIEW

File: tools/node_tcp.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+const port = process.argv[2] || "4544";
+console.log("port", port);
+
+const response = Buffer.from(
+  "HTTP/1.1 200 OK\r\nContent-Length: 12\r\n\r\nHello World\n"
+);
+
+Server(socket => {
+  socket.on("data", _ => {
+    socket.write(response);
```
Comment (piscisaureus) [inline tools/node_tcp.js]: should be socket.end() ? Otherwise this leaks FDs like there's no tomorrow.

Comment (piscisaureus) [inline tools/node_tcp.js]: Should be `.end()` or this leaks sockets?
Or are you just profiling one tcp stream?

Comment (ry) [reply tools/node_tcp.js]: we're profiling keep-alive - so it shouldn't end here.

Comment (kyranet) [reply tools/node_tcp.js]: The command we use to benchmark - `third_party/wrk/linux/wrk -d 10s http://127.0.0.1:4544/`, creates 10 connections, thus 10 sockets are created to reply simultaneously (thought in practise this is creating 11). We can confirm this by adding those lines:

```diff
+ let i = 0;
  Server(socket => {
+   console.log(`Connected to socket no ${++i}`)
    socket.on("data", _ => {
    socket.write(response);
```

Which will log:

```
Connected to socket no 1
Connected to socket no 2
Connected to socket no 3
Connected to socket no 4
Connected to socket no 5
Connected to socket no 6
Connected to socket no 7
Connected to socket no 8
Connected to socket no 9
Connected to socket no 10
Connected to socket no 11
```

Comment (hayd) [reply tools/node_tcp.js]: > we're profiling keep-alive - so it shouldn't end here.

Perhaps a comment to this effect will prevent it being "fixed" later.

Comment (piscisaureus) [reply tools/node_tcp.js]: I second @hayd 's suggestion.

**Comment by piscisaureus** [comment]
@ry I tentatively approved it but I would suggest to add a comment explaining why the socket is never closed.

File: tools/node_tcp.js (hunk 1 above)
Comment (ry) [reply tools/node_tcp.js]: Fixed.

**Pull request merged**